proxy/src/http/source.rs

```diff
@@ -0,0 +1,1054 @@
+//! Endpoints and serialisation for source code browsing.
+
+use librad::paths::Paths;
+use serde::ser::SerializeStruct as _;
+use serde::{Deserialize, Serialize, Serializer};
+use std::sync::Arc;
+use tokio::sync::RwLock;
+use warp::document::{self, ToDocumentedType};
+use warp::{path, Filter, Rejection, Reply};
+
+use crate::coco;
+
+/// Prefixed filters.
+pub fn routes(
+    paths: Arc<RwLock<Paths>>,
```

Comment:
Why is this in a lock? Are we writing to it?

Reply:
Esp. during testing we reset/replace the path with a new temp location, to nuke the states.